lib/src/widget/nav_bar/utils/observer.dart

```diff
@@ -0,0 +1,30 @@
+import 'package:flutter/material.dart';
+import 'package:xayn_design/src/widget/nav_bar/widget/nav_bar_container.dart';
+
+class NavBarObserver extends NavigatorObserver {
+  @override
+  void didPop(Route route, Route? previousRoute) {
+    _resetNavBarConfig(true);
+  }
+
+  @override
+  void didPush(Route route, Route? previousRoute) {
+    _resetNavBarConfig(false);
+  }
+
+  @override
+  void didRemove(Route route, Route? previousRoute) {
+    _resetNavBarConfig(true);
+  }
+
+  @override
+  void didReplace({Route? newRoute, Route? oldRoute}) {
+    _resetNavBarConfig(false);
+  }
+
+  void _resetNavBarConfig(bool goingBack) {
+    final context = navigator?.context;
+    if (context == null) return;
+    NavBarContainer.resetNavBar(context, goingBack: goingBack);
```

Comment:
❗  I think this should not be a singleton: better NavBarContainer.of(context)

Reply:
again, `.of(context)` is also static function.
and cos this widget lives on top of all screens - yes, it is like a singleton (in fact no)